runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/DefaultJobBundleFactory.java

```diff
@@ -161,29 +166,33 @@ public StageBundleFactory forStage(ExecutableStage executableStage) {
           ProcessBundleDescriptors.fromExecutableStage(
               stageIdGenerator.getId(),
               executableStage,
-              dataServer.getApiServiceDescriptor(),
-              stateServer.getApiServiceDescriptor());
+              wrappedClient.getServerInfo().getDataServer().getApiServiceDescriptor(),
+              wrappedClient.getServerInfo().getStateServer().getApiServiceDescriptor());
     } catch (IOException e) {
       throw new RuntimeException(e);
     }
-    return SimpleStageBundleFactory.create(wrappedClient, processBundleDescriptor, stateServer);
+    return SimpleStageBundleFactory.create(
+        wrappedClient, processBundleDescriptor, wrappedClient.getServerInfo().getStateServer());
   }
 
   @Override
   public void close() throws Exception {
+    // Tear down common servers.
```

Comment:
Is there a reason why the tear down order was changed? (Previously environment was closed before closing services.) I seem to remember that the order mattered.